jetty-server/src/main/java/org/eclipse/jetty/server/Request.java

```diff
@@ -498,57 +498,30 @@ public void extractFormParameters(MultiMap<String> params)
     {
         try
         {
-            int maxFormContentSize = -1;
-            int maxFormKeys = -1;
+            int maxFormContentSize = ContextHandler.DEFAULT_MAX_FORM_CONTENT_SIZE;
+            int maxFormKeys = ContextHandler.DEFAULT_MAX_FORM_KEYS;
 
             if (_context != null)
             {
-                maxFormContentSize = _context.getContextHandler().getMaxFormContentSize();
-                maxFormKeys = _context.getContextHandler().getMaxFormKeys();
+                ContextHandler contextHandler = _context.getContextHandler();
+                maxFormContentSize = contextHandler.getMaxFormContentSize();
+                maxFormKeys = contextHandler.getMaxFormKeys();
             }
-
-            if (maxFormContentSize < 0)
+            else
```

Comment:
Note that this is incompatible as seen in https://github.com/jenkinsci/winstone/pull/95/commits/8a8a146c88e55e2258e063a041fb43c21753cf5b.

Reply:
Opened Issue #4638